### PR TITLE
Add validation to article model schema

### LINKED ISSUE #28: Add validation to Article Model schema
[See](https://mongoosejs.com/docs/validation.html)

### PR DESCRIPTION
**Pull request by Asbelll**
Closes #28 

- Creates validation for the SCHEMA class, should not be needed as we are now but a "just in case" layer of security
- Updates the maximum allowed number of characters for the article title on the dto
<br />

Todo:
- Catch 400 errors
- Change the result.success on the update method in the service (should have a way to distinguish if the error comes from a not found or invalid data)
<br />

As it is, the error we're catching for the SCHEMA validation is the entire error.errors object, not very user friendly but really easy to change if it's ever needed

### CODE REVIEW

File: src/app/models/article.model.js [hunk 1]
```diff
@@ -3,11 +3,32 @@ import { Schema } from 'mongoose'
...
     static SCHEMA = new Schema({
-        title: String,
-        body: String,
-        tags: [String],
+        title: {
+            type: String,
+            minLength: [3, "Title must have at least 3 characters"],
+            maxLength: 250
+        },
+        body: {
```
Comment (MtsReis) [inline src/app/models/article.model.js:12]: Title and body should be [required](https://mongoosejs.com/docs/api/schematype.html#SchemaType.prototype.required()).

File: src/app/models/article.model.js [hunk 2]
```diff
@@ -3,11 +3,32 @@ import { Schema } from 'mongoose'
...
+        tags: {
+            type: [String],
+            validate: [(value) => {
+                for (let k in value) {
+                    if (value[k].length < 2) {
+                        return false
+                    }
+                }
+                return value.length > 0 && value.length <= 10
+            }, 'Needs at least 1 tag, max of 10 and must have at least 2 characters']
```
Comment (MtsReis) [inline src/app/models/article.model.js]: There shouldn't be a minimum amount of tags.
I like how it thoroughly validates this property btw.

**Review by MtsReis — DISMISSED** [review]
Looks pretty good.
Approved. If possible, apply the changes suggested before merging it, please. :smile:

**Comment by Asbelll** [comment]
Added 'required' parameter to title and body.
Tags should not be a problem even if left untouched since the validate method only runs when the field is filled, but removed the "must be higher than 0" check since it would be redundant.

**Review by MtsReis — APPROVED** [review]

**Pull request merged**